reader.go

```diff
@@ -783,6 +787,55 @@ func (r *Reader) commitLoop(conn *Conn) func(stop <-chan struct{}) {
 	}
 }
 
+// partitionWatcher queries kafka and watches for partition changes, triggering a rebalance if changes are found.
+func (r *Reader) partitionWatcher(conn *Conn) func(stop <-chan struct{}) {
+	return func(stop <-chan struct{}) {
+		ticker := time.NewTicker(defaultPartitionWatchTime)
+		defer ticker.Stop()
+		ops, err := conn.ReadPartitions(r.config.Topic)
+		if err != nil {
+			r.withErrorLogger(func(l *log.Logger) {
+				l.Printf("Problem getting partitions during startup, %v\n, Returning and setting up handshake", err)
+			})
+			return
+		}
+		// It's possible that the list of partitions returned are not in any order
+		// so put them into a map to compare.
+		oParts := make(map[int]struct{})
+		for _, p := range ops {
+			oParts[p.ID] = struct{}{}
+		}
+		for {
+			select {
+			case <-stop:
+				return
+			case <-ticker.C:
+				ops, err := conn.ReadPartitions(r.config.Topic)
+				if err != nil {
+					r.withErrorLogger(func(l *log.Logger) {
+						l.Printf("Problem getting partitions while checking for changes, %v\n", err)
+					})
+					return
+				}
+				if len(ops) != len(oParts) {
+					r.withErrorLogger(func(l *log.Logger) {
+						l.Printf("Partition changes found, reblancing group: %v.", r.config.GroupID)
+					})
+					return
+				}
+				for _, p := range ops {
```

Comment:
Do we need this check here?  Seems to me like the length check above should be sufficient.  I'm having a hard time imagining a case where this check would cause us to return where the length check would not.

Reply:
I agree. Just checking the length should work.